core/ra2.js

```diff
@@ -0,0 +1,259 @@
+import "../vendor/aes.js";
+import "../vendor/sha1.js";
+import "../vendor/jsencrypt.js";
+import Base64 from './base64.js';
+
+export class RA2Cipher {
```

Comment:
Given that you've figured out the algorithm used, can this be reorganised so it follows the same naming and structure as the algorithm's specifications?